src/data/reports/data-quality/DataQualityDefaultRepository.ts

```diff
@@ -0,0 +1,276 @@
+        };
+        return { pager: pager, objects: dataQualityIndicatorErrorsInPage };
+    }
+
+    async saveDataQuality(namespace: string, dataQuality: IndicatorConfig | ProgramIndicatorConfig): Promise<void> {
+        return await this.globalStorageClient.saveObject<IndicatorConfig | ProgramIndicatorConfig>(
+            namespace,
+            dataQuality
+        );
+    }
+
+    async reloadValidation(namespace: string, fromZero: boolean) {
+        const dataQuality = await this.globalStorageClient.getObject<IndicatorConfig | ProgramIndicatorConfig>(
+            namespace
+        );
+        const dataQualityErrors: any[] = [];
+
+        if (fromZero && dataQuality?.validationResults.length === 0) {
+            const { indicators } = await this.api.metadata
+                .get({
+                    indicators: {
+                        fields: {
+                            id: true,
+                            name: true,
+                            numerator: true,
+                            denominator: true,
+                            lastUpdated: true,
+                            user: true,
+                        },
+                    },
+                })
+                .getData();
+            const { programIndicators } = await this.api.metadata
+                .get({
+                    programIndicators: {
+                        fields: {
+                            id: true,
+                            name: true,
+                            numerator: true,
+                            denominator: true,
+                            lastUpdated: true,
+                            user: true,
+                            expression: true,
+                            filter: true,
+                        },
+                    },
+                })
+                .getData();
+
+            await promiseMap(indicators, async indicator => {
+                const dataQualityItem: DataQualityItem = indicator;
+                try {
+                    const numeratorValidation = await this.api.expressions
+                        .validate("indicator", indicator.numerator)
+                        .getData();
+                    const denominatorValidation = await this.api.expressions
+                        .validate("indicator", indicator.denominator)
+                        .getData();
+
+                    if (numeratorValidation.status === "ERROR" || denominatorValidation.status === "ERROR") {
+                        dataQualityItem.numeratorresult = numeratorValidation.status === "ERROR";
+                        dataQualityItem.denominatorresult = denominatorValidation.status === "ERROR";
+                        dataQualityItem.metadataType = "Indicator";
+                        dataQualityErrors.push(dataQualityItem);
+                    }
+                } catch (error) {
+                    console.debug(error);
+                }
+            });
```

Comment:
This chunk of code gets reused verbatim below, the same happens with the `promiseMap` bellow.